stl/inc/xstring

```diff
@@ -3980,6 +3980,21 @@ public:
         }
     }
 
+#if _HAS_CXX23
+    template <class _Operation>
+    constexpr void resize_and_overwrite(_CRT_GUARDOVERFLOW const size_type _Newsize, _Operation _Op) {
+        if (_Mypair._Myval2._Myres < _Newsize) {
+            _Reallocate_grow_by(
+                _Newsize - size(),
+                [](_Elem* const _New_ptr, const _Elem* const _Old_ptr, const size_type _Oldsize,
+                    const size_type _Newsize) { _Traits::copy(_New_ptr, _Old_ptr, (_STD min) (_Oldsize, _Newsize)); },

```

Comment:
Why are we performing a copy when we are reallocating?

During constant evaluation we must start the lifetime of the new elements, but otherwise I see no reason to write something that will be overwritten in the next step

Reply:
It is required:
> p be a charT*, such that the range [p, p + n] is valid and this->compare(0, k, p, k) == 0 is true before the call. The values in the range [p + k, p + n] may be indeterminate